### PR TITLE
Fix version parsing

### PR DESCRIPTION
**Pull request by folmos-at-orange**

### CODE REVIEW

File: khiops/core/internals/version.py [hunk 1]
```diff
@@ -73,6 +73,11 @@ def __init__(self, version_str):
                     "PATCH-PRE_RELEASE version part must contain a single '-'",
                     version_str,
                 )
+            if patch_and_pre_release.count(".") > 1:
+                self._raise_init_error(
+                    "PATCH-PRE_RELEASE version part must at most a single '.'",
```
Comment (popescu-v) [inline khiops/core/internals/version.py:78]: s/must at most/must contain at most/

Comment (folmos-at-orange) [reply khiops/core/internals/version.py:78]: Ok

File: khiops/core/internals/version.py [hunk 2]
```diff
@@ -123,12 +132,12 @@ def patch(self):
...
         """str : The version's pre-release tag
 
-        Returns: either 'a', 'b' or 'rc' followed by a number or None.
+        Returns: either 'a', 'b' or 'rc' followed by '.' and a number or None.
         """
         if self._pre_release_id is None:
             return None
         else:
-            return f"{self._pre_release_id}{self._pre_release_increment}"
+            return f"{self._pre_release_id}.{self._pre_release_increment}"
```
Comment (popescu-v) [inline khiops/core/internals/version.py:140]: What if `self.pre_release_increment` is empty, say, for a pre-release of the form (ill-formed): `.a` (error) or `.a.` (typo)?

Comment (folmos-at-orange) [reply khiops/core/internals/version.py:140]: This is checked above. If there are more than one `.` in the pre-release particle it fails to create the version object.

File: khiops/core/internals/runner.py [hunk 3]
```diff
@@ -741,74 +741,60 @@ def _report_exit_status(
...
+            # Raise an exception if there were errors
+            if errors or fatal_errors or return_code != 0:
+                raise KhiopsRuntimeError(
+                    f"{tool_name} execution had errors (return code {return_code}):\n"
+                    f"{error_msg}"
+                )
+            # Otherwise show the message as a warning
+            else:
+                error_msg = (
+                    f"Khiops ended correctly but there were minor issues:\n{error_msg}"
```
Comment (popescu-v) [inline khiops/core/internals/runner.py:795]: I would remove the word "minor" as here we also collect errors and _fatal_ errors, even though Khiops returned correctly. I would just say: "but there were some issues" and leave the user to appreciate whether they are minor or not.

Comment (folmos-at-orange) [reply khiops/core/internals/runner.py:795]: If there are errors (fatal or not) this branch is not executed.

File: khiops/core/internals/version.py [hunk 4]
```diff
@@ -36,12 +36,12 @@ def __init__(self, version_str):
         self._version_str = version_str
 
         # Remove the "v" prefix if present
-        raw_parts = re.sub("^v", "", self._version_str).split(".")
+        raw_parts = re.sub("^v", "", self._version_str).split(".", maxsplit=2)
 
         # Check the Khiops version format: MAJOR.MINOR.PATCH[-PRE_RELEASE]
         if len(raw_parts) != 3:
```
Comment (popescu-v) [inline khiops/core/internals/version.py:42]: Given the `maxsplit`, we can now check `if len(raw_parts) < 3`

File: khiops/core/internals/version.py (hunk 1 above)
Comment (folmos-at-orange) [reply khiops/core/internals/version.py:78]: Done

File: tests/test_core.py [hunk 5]
```diff
@@ -2198,6 +2199,7 @@ def test_invalid_versions(self):
             "10.0",
             "10i.4.0",
             "10.4b.3",
+            "10.4.1-b..2",
```
[the remark this answers is not in the document]
Comment (folmos-at-orange) [reply tests/test_core.py:2202]: Done

File: khiops/core/internals/version.py (hunk 2 above)
Comment (popescu-v) [reply khiops/core/internals/version.py:140]: Yes, but the situation where the prerelease has the form `a.`, with a trailing dot without the increment, is it handled somewhere?

Comment (folmos-at-orange) [reply khiops/core/internals/version.py:140]:
```
(06:26:39)> python -c 'from khiops.core.internals.version import KhiopsVersion; KhiopsVersion("10.1.1-b.")'
Traceback (most recent call last):
  File "<string>", line 1, in <module>
  File "/Users/folmos/work/khiops/khiops-python/khiops/core/internals/version.py", line 109, in __init__
    self._raise_init_error(
  File "/Users/folmos/work/khiops/khiops-python/khiops/core/internals/version.py", line 114, in _raise_init_error
    raise ValueError(f"{msg}. Version string: '{version_str}'.")
ValueError: PRE_RELEASE version part increment is not numeric. Version string: '10.1.1-b.'.
```

File: khiops/core/internals/version.py (hunk 4 above)
Comment (folmos-at-orange) [reply khiops/core/internals/version.py]: Ok & Done

**Pull request merged**